src/librustdoc/html/static/main.js

```diff
@@ -428,6 +428,15 @@ function hideThemeButtonState() {
         handleHashes(ev);
     }
 
+    function openParentDetails(elem) {
+        while (elem) {
```

Comment:
Shouldn't there be a "better" stop condition here? Otherwise it'll go up to the top of the hierarchy which isn't really an issue but seems like a waste.

Reply:
FWIW, going up to the top of the hierarchy was my recommendation in https://github.com/rust-lang/rust/pull/84321#discussion_r615487368. My thinking is:

 - this reduces the JS code's dependence on the exact structure of the HTML
 - our HTML is only 16 levels deep at max (last I checked)
 - this only gets called once during load, when there is a URL fragment present

Reply:
My only concern is: can it impact strongly performance like DOM iterators (`getElementsByClassName` for example)? That was making terrible performance issues on chrome and led to the creation of the `onEachLazy` function.

Ohterwise I agree with you: I don't want this code to be specific to the HTML layout.

Reply:
Good point about performance. I _think_ it shouldn't be a big deal, because in almost every case there will be at most one parent `<details>`. As I understand it, the performance issue with DOM iterators comes when you modify the DOM while iterating. That causes Chrome to recalculate the iterator, which can be expensive. The general rule here is "read, then modify." So if we wanted to fine-tune the performance here, we could collect all the `<details>` we find walking up the tree, and once we're finished walking, set them all to open. But again, I suspect the performance impact is minimal in this particular situation.

Reply:
I can confirm that we weren't modifying the iterator, simply going through it. It was terrible so now I'm very cautious about such thing. If I can get a simple bench comparing before/after, I'll happily agree. Numbers are absolute. :)

Reply:
I did a benchmark and found that this takes about 75 _micro_seconds in Chrome. In Firefox, the performance timer is millisecond resolution, and just shows this as 0.

```
function time(f) {
  var begin = performance.now();
  f();
  console.log("elapsed", performance.now() - begin)
}
function climb(el) {
  while (el.parentNode) {
    if (el.tagName === "DETAILS") {
      el.open = true;
    }
    el = el.parentNode
  }
}
time(() => climb(temp1))
elapsed 0.07500004721805453
```

temp1 is assigned by visiting https://doc.rust-lang.org/std/string/struct.String.html#impl-Add%3C%26%27_%20str%3E, highlighting the "Add" node in the inspector, and assigning it to a variable.

Reply:
Then it's more than acceptable. Thanks for taking the time to check!